source/code/source/playfab/PlayFabEvent.cpp

```diff
@@ -89,6 +91,8 @@ namespace PlayFab
         this->eventContents.Payload[name] = value;
 #elif defined(PLAYFAB_PLATFORM_ANDROID)
         this->eventContents.Payload[name] = (Uint64)value;
+#elif defined(JSON_HAS_INT64)
```

Comment:
>JSON_HAS_INT64 [](start = 14, length = 14)

this elif feels super weird

the other guards are based on platform but this one is  feature flag? what happens if a platform like playstation or android also has this defined? it will overwrite the payload twice and you may not get what you expect.

It also doesn't make a lot of sense to me why these ifdef guards are event necessary in the first place. Why can't all platforms be set to the raw u64 value instead of 3 of them requiring casts?

Reply:
I agree.
I'll file a follow-up bug for this, because I'm rather desperate to get this checked in, as it's tested and working.